### ISSUE TITLE
Unexpected behaviour for "ng build --aot"

### ISSUE THREAD

**Issue by DibyodyutiMondal** [angular/angular-cli#14709]
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?
Can't say for sure

### Description
```
<mat-form-field appearance="outline">
     <mat-label>User Type</mat-label>
     <mat-select [(value)]="type">
          <mat-option *ngFor="let role of typeOptions"
                                [value]="role">
             {{role}}
          </mat-option>
     </mat-select>
</mat-form-field>
```
The above code is present in a template for a component inside a lazy loaded module put behind canLoad and canActivate guards
The app is designed to not to fetch any lazy-loaded modules except the login module, which sets up the current user from a jwt, etc etc, and only after that the canLoad guards will allow other modules to be fetched.

Everything works fine if I use **ng build --watch** 
However, if I use **ng build --watch --aot**, then for some reason the server (IIS) returns a 404
for the chunk 'default~.......-~(somenumber).js', even though the file is already present in the file system folder.
The CLI generates no errors, either.
Obviously, this renders my entire app useless, because unless I log in, I can't do anything at all.

There are no problems if I use **ng build --prod**

I can't even diagnose where the error is happening exactly, as the modules are not even loading
The only difference between this and other similar implementations of mat-select in my app is that the mat-options in this case are made from an array of string enumerations (typeOptions).

## 🌍 Your Environment
<pre><code>

Angular CLI: 8.0.2
Node: 12.2.0
OS: win32 x64
Angular: 8.0.0
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, material, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.800.2
@angular-devkit/build-angular     0.800.2
@angular-devkit/build-optimizer   0.800.2
@angular-devkit/build-webpack     0.800.2
@angular-devkit/core              8.0.2
@angular-devkit/schematics        8.0.2
@angular/cli                      8.0.2
@ngtools/webpack                  8.0.2
@schematics/angular               8.0.2
@schematics/update                0.800.2
rxjs                              6.4.0
typescript                        3.4.5
webpack                           4.30.0

</code></pre>

**Other Environment Info**
Browser:Chrome 74
IDE: Visual Studio Community 2019 16.2 Preview 1
Server: IIS 10

**Comment by istiti** [comment]
Please can you make a repro?
You use  `canLoad` and `canActivate` together, why?

**Comment by DibyodyutiMondal** [comment]
I use canLoad and canActivate together because of a specific case where say, an admin logged into the site, did his work and then logged out, and another user with lesser privileges uses the website on the same machine, the admin's module chunk would already be present in the cache. After going through numerous stackOverflow questions and the docs, it was unclear to me whether the canLoad guard would prevent the admin module from being used if it's chunk is already in the cache. So I use canActivate along with canLoad, just to stay on the safe side.
Although, I'll admit, I don't have in-depth knowledge of how angular handles the cache, and neither did I test out my theory, because I haven't yet got to the part where I add the lesser-privileged users. Once I do, I'll be able to test my theory myself.

As for the repro, I'll publish asap

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. However, you didn't provide sufficient information for us to understand and reproduce the problem. 

Can you setup a minimal repro please?

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

**Comment by DibyodyutiMondal** [comment]
I am trying to create a repro, but I am unable to reproduce the error, ie, the chunks load both with and without '--aot'

As for the folder structure, it is true that my directory structure is rather... unconventional.
Could this issue please be kept open while I restructure my project? (2-3 days) at the most.
I'll report back as soon as I am done.

**Comment by DibyodyutiMondal** [comment]
![mat-form-field Error](https://user-images.githubusercontent.com/50345677/59241762-8b95b180-8c26-11e9-9414-8bd6accc2d52.png)

In the meantime, I am uploading a screenshot showing my folder structure

**Comment by DibyodyutiMondal** [comment]
Sorry. Closed by mistake.

**Comment by mgechev** [comment]
@DibyodyutiMondal this looks like an environment specific issue. Maybe not a direct problem in the CLI? Let us know if you're able to create a minimal demo where we can reproduce the problem.

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. However, you didn't provide sufficient information for us to understand and reproduce the problem. Please check out [our submission guidelines](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) to understand why we can't act on issues that are lacking important information.

If the problem persists, please file a new issue and ensure you provide all of the required information when filling out the issue template.
